Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -435,24 +437,31 @@ def _addSampleLogInfo(self, ws, linePosition):
 
     def _moveDetector(self, ws, linePosition):
         """Perform detector position correction for direct and reflected beams."""
-        if self.getProperty(Prop.DIRECT_LINE_WORKSPACE).isDefault:
-            return ws
         # Reflected beam calibration
+        if self.getProperty(Prop.DIRECT_LINE_WORKSPACE).isDefault and \
```

Comment:
The only time we want to avoid moving the detector is

```python
if not self.getProperty(Prop.LINE_POSITION_INPUT).isDefault and not self.getProperty(Prop.TWO_THETA).isDefault and self.getProperty(Prop.INPUT_WS).isDefault
```

In this case the loader should have done it for us, see lines 367 and 373.

Reply:
This condition breaks actually the unit tests for the default run (D17 and Figaro)